src/Morphir/IR/Type/DataCodec.elm

```diff
@@ -414,6 +449,73 @@ decodeData ir tpe =
+                                        apply v
+                                            |> Interpreter.evaluate SDK.nativeFunctions ir
+                                            |> (\evaluatedResult ->
+                                                    case evaluatedResult of
+                                                        Ok val ->
+                                                            -- if the val is of the result or maybe type,
+                                                            -- we need to extract the value
+                                                            case val of
+                                                                -- if the evaluated value is "Just value"
+                                                                Value.Apply _ (Value.Constructor _ ( [ [ "morphir" ], [ "s", "d", "k" ] ], [ [ "maybe" ] ], [ "just" ] )) value ->
+                                                                    Decode.succeed value
+
+                                                                -- if the evaluated value "Nothing"
+                                                                Value.Constructor _ ( [ [ "morphir" ], [ "s", "d", "k" ] ], [ [ "maybe" ] ], [ "nothing" ] ) ->
+                                                                    fnName
+                                                                        ++ " returned Nothing"
+                                                                        |> Decode.fail
+
+                                                                -- if the evaluated value is "Ok value"
+                                                                Value.Apply _ (Value.Constructor _ ( [ [ "morphir" ], [ "s", "d", "k" ] ], [ [ "result" ] ], [ "ok" ] )) value ->
+                                                                    Decode.succeed value
+
+                                                                -- if the evaluated value "Err err"
+                                                                Value.Apply _ (Value.Constructor _ ( [ [ "morphir" ], [ "s", "d", "k" ] ], [ [ "result" ] ], [ "err" ] )) err ->
+                                                                    let
+                                                                        _ =
+                                                                            Debug.log "Could not derive value" err
+                                                                    in
+                                                                    fnName
+                                                                        ++ " returned Err"
+                                                                        |> Decode.fail
+
+                                                                -- only Maybe and Result types are expected, we can fail
+                                                                _ ->
```

Comment:
I decided to fail here if a `Maybe` or `Result` Type wasn't returned from the evaluation.
Do I need to change this approach and allow for other values or is it okay to enforce this standard at this time?